### PR TITLE
📫 Replace flowserver completely with mailroom

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,18 +14,27 @@ addons:
...
-env:
-  - ES_VERSION=6.3.1 ES_DOWNLOAD_URL=https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-${ES_VERSION}.tar.gz
-
 cache:
   directories:
     - "$HOME/.cache/pip"
 
 install:
-  - wget ${ES_DOWNLOAD_URL}
+  # install Elasticsearch
```
Comment (rowanseymour) [inline .travis.yml]: just making this consistent - not much advantage to using `env` unless you're building a matrix

File: .travis.yml [hunk 2]
```diff
@@ -85,8 +82,8 @@ script:
   - python manage.py collectstatic --noinput --verbosity=0
   - (! python manage.py compress --extension=".haml" --settings=temba.settings_compress | grep 'Error') || exit 1
 
-  # fire up goflow
-  - ./flowserver --log-level=warn &
+  # fire up mailroom
+  - ./mailroom --log-level=warn &
```
Comment (rowanseymour) [inline .travis.yml]: @nicpottier not sure what to do here - mailroom fires up - throws out some errors about the database being wrong - but the bits we need still work. I'm concerned if I configure it to see the real test database, then it's going to do stuff we don't want during tests.

File: temba/flows/models.py [hunk 3]
```diff
@@ -87,6 +86,32 @@
 START_FLOW_BATCH_SIZE = 500
 
 
+class Events(Enum):
```
Comment (rowanseymour) [inline temba/flows/models.py]: this is currently only used for identifying msg events (by name so the numbers aren't important) for things like message exports

File: temba/flows/server/mailroom.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    def sim_start(self, payload):
+        return self._request("sim/start", payload)
+
+    def sim_resume(self, payload):
+        return self._request("sim/resume", payload)
+
+    def _request(self, endpoint, payload):
+        if self.debug:
+            print("[MAILROOM]=============== %s request ===============" % endpoint)
```
[the remark this answers is not in the document]
Comment (rowanseymour) [reply temba/flows/server/mailroom.py]: yeah I copied this code from the previous client class but I'll update it to use logging

File: .travis.yml (hunk 2 above)
Comment (nicpottier) [reply .travis.yml]: It isn't going to do anything (yet) unless an org / flow has flow_server_enabled set to true. You will need the DB for simulation as it has to load assets etc, so probably need to set it up right.

Comment (rowanseymour) [reply .travis.yml]: that true even for campaign events?

Comment (nicpottier) [reply .travis.yml]: yep: https://github.com/nyaruka/mailroom/blob/master/campaigns/cron.go#L191

Comment (rowanseymour) [reply .travis.yml]: ah ok I thought we were just treating all `is_system=True` flows as effectively `flow_server_enabled` - I just noticed that new campaign event flows aren't having `flow_server_enabled` set so these are actually going through the old engine. We switched existing ones over in a migration a while ago but didn't change how new ones are created.

Comment (rowanseymour) [reply .travis.yml]: kinda funny cuz I now have RapidPro unit tests and mailroom running from a same database and Redis... but I have a failing test (`test_mailroom_starts`) which expects something to be left in a Redis queue but apparently mailroom is picking it up and handling it

**Pull request merged**